staging/src/k8s.io/client-go/util/jsonpath/jsonpath_test.go

```diff
@@ -286,6 +286,8 @@ func TestKubernetes(t *testing.T) {
 		{"hostname", `{.items[0].metadata.labels.kubernetes\.io/hostname}`, &nodesData, "127.0.0.1", false},
 		{"hostname filter", `{.items[?(@.metadata.labels.kubernetes\.io/hostname=="127.0.0.1")].kind}`, &nodesData, "None", false},
 		{"bool item", `{.items[?(@..ready==true)].metadata.name}`, &nodesData, "127.0.0.1", false},
+		{"last index", `{.items[-1].metadata.name}`, &nodesData, "127.0.0.2", false},
```

Comment:
As far as I can tell, for JSON Path, `[-1:]` should return a list with just the last item, but `[-1]` is an error.

* https://goessner.net/articles/JsonPath/ says that `[]` is the native js array index operator
* The native js array index operator does not appear to allow negative indexing.

So, I think this should not work. `.items[(@.length-1)]` is the correct syntax.

Reply:
(sorry for slow response, I didn't see this until now)

Reply:
I have changed this value.